utils/s2n_blob.c

```diff
@@ -47,3 +47,36 @@ int s2n_blob_char_to_lower(struct s2n_blob *b)
 
     return 0;
 }
+
+/* similar to s2n_stuffer_alloc_ro_from_hex_string() but for a byte array */
+int s2n_hex_string_to_byte(const char* str, uint8_t *target)
+{
+    for (int i = 0; i < strlen(str); i += 2) {
+        uint8_t u = 0;
+
+        if (str[i] >= '0' && str[i] <= '9') {
```

Comment:
Have you looked at how we do base64 with tables? Tables end up being far simpler here.